File: Example/Tests/NYTPhotoCaptionViewTests.m
//
//  NYTPhotoCaptionViewTests.m
//  NYTPhotoViewer
//
//  Created by Brian Capps on 2/24/15.
//  Copyright (c) 2015 NYTimes. All rights reserved.
//

@import UIKit;
@import XCTest;

#import <NYTPhotoViewer/NYTPhotoCaptionView.h>

@interface NYTPhotoCaptionViewTests : XCTestCase

@end

@implementation NYTPhotoCaptionViewTests

- (void)testDesignatedInitializerAcceptsNilForTitle {
    XCTAssertNoThrow([[NYTPhotoCaptionView alloc] initWithAttributedTitle:nil attributedSummary:[[NSAttributedString alloc] init] attributedCredit:[[NSAttributedString alloc] init]]);
}

- (void)testDesignatedInitializerAcceptsNilForSummary {
    XCTAssertNoThrow([[NYTPhotoCaptionView alloc] initWithAttributedTitle:[[NSAttributedString alloc] init] attributedSummary:nil attributedCredit:[[NSAttributedString alloc] init]]);
}

- (void)testDesignatedInitializerAcceptsNilForCredit {
    XCTAssertNoThrow([[NYTPhotoCaptionView alloc] initWithAttributedTitle:[[NSAttributedString alloc] init] attributedSummary:[[NSAttributedString alloc] init] attributedCredit:nil]);
}

@end

File: Example/NYTPhotoViewer-SwiftTests/NYTPhotoCaptionViewTests.swift
//
//  NYTPhotoCaptionViewTests.swift
//  NYTPhotoViewer
//
//  Created by Mark Keefe on 3/21/15.
//  Copyright (c) 2015 The New York Times. All rights reserved.
//

import XCTest

class NYTPhotoCaptionViewTests: XCTestCase {

    func testDesignatedInitializerAcceptsNilForTitle() {
        let photoCaptionView = NYTPhotoCaptionView(attributedTitle: nil, attributedSummary: NSAttributedString(), attributedCredit: NSAttributedString())
        XCTAssertNotNil(photoCaptionView)
    }
    
    func testDesignatedInitializerAcceptsNilForSummary() {
        let photoCaptionView = NYTPhotoCaptionView(attributedTitle: NSAttributedString(), attributedSummary: nil, attributedCredit: NSAttributedString())
        XCTAssertNotNil(photoCaptionView)
    }
    
    func testDesignatedInitializerAcceptsNilForCredit() {
        let photoCaptionView = NYTPhotoCaptionView(attributedTitle: NSAttributedString(), attributedSummary: NSAttributedString(), attributedCredit: nil)
        XCTAssertNotNil(photoCaptionView)
    }
}
